### DISCUSSION TITLE
Blank space for :path in path parameter

### DISCUSSION THREAD

**Discussion by jamfie** [fastapi/fastapi#7186] [Questions] [answered]
Hi. Was going through the [documentation](https://fastapi.tiangolo.com/tutorial/path-params/) regarding path parameters. I realize it works when

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get('/files/{file_path:path}')
async def read_file(file_path: str):
    return({"file_path": f_path})
```

but return 404 when

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get('/files/{file_path: path}')
async def read_file(file_path: str):
    return({"file_path": f_path})
```

Notice how there is a blank space after the colon for the wrong demonstration: **file_path: path**. Is this behavior expected?

**Comment by coder3112** [comment, marked as answer]
That is because it considers the space as a character and wants you to put that extra space in the URL which is not possible. So i suggest that you stick with the first code, why change it when it works.

**Comment by raphaelauv** [comment]
@jamfie close the issue if it's alright , thank you

**Comment by tiangolo** [comment]
Thanks for the help here @coder3112 and @raphaelauv! 🍰 

And thanks for closing it @jamfie ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
